### PR TITLE
fix(ext/node): implement asymmetric keygen

### PR DESCRIPTION
**Pull request by littledivy**
Towards #18455 

This commit implements the keypair generation for asymmetric keys for the `generateKeyPair` API.

See how key material is managed in this implementation: https://www.notion.so/denolandinc/node-crypto-design-99fc33f568d24e47a5e4b36002c5325d?pvs=4

Private and public key encoding depend on `KeyObject#export` which is not implemented. I've also skipped ED448 and X448 since we need a crate for that in WebCrypto too.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -3295,9 +3326,9 @@ version = "0.4.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "eff33bdbdfc54cc98a2eca766ebdec3e1b8fb7387523d5c9c9a2891da856f719"
 dependencies = [
- "der",
- "pkcs8",
- "spki",
+ "der 0.6.1",
+ "pkcs8 0.9.0",
+ "spki 0.6.0",
```
Comment (littledivy) [inline Cargo.lock:3331]: uh oh. hold up

**Review by bartlomieju — COMMENTED** [review]
Implementation wise it looks great - I assume the native Node.js tests still can't be enabled due to other missing APIs. I'm worried about so many duplicate versions of dependencies being used though.

File: .github/workflows/ci.generate.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ const Runners = (() => {
 })();
 // bump the number at the start when you want to purge the cache
 const prCacheKeyPrefix =
-  "18-cargo-target-${{ matrix.os }}-${{ matrix.profile }}-${{ matrix.job }}-";
+  "19-cargo-target-${{ matrix.os }}-${{ matrix.profile }}-${{ matrix.job }}-";
```
[the remark this answers is not in the document]
Comment (littledivy) [reply .github/workflows/ci.generate.ts:20]: ci link issues with the cache

**Review by bartlomieju — APPROVED** [review]

File: Cargo.lock (hunk 1 above)
Comment (littledivy) [reply Cargo.lock:3343]: we need to upgrade all the RustCrypto deps. I'll do it after this lands.

**Pull request merged**